### PR TITLE
New package: xremap-0.4.4

### PR DESCRIPTION
**Pull request by akhiljalagam**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements): **YES**

### CODE REVIEW

File: srcpkgs/xremap/template [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+maintainer="akhil <akhil.j@outlook.in>"
+license="MIT"
+homepage="https://github.com/k0kubun/xremap"
+changelog="https://raw.githubusercontent.com/k0kubun/xremap/master/CHANGELOG.md"
+distfiles="https://github.com/k0kubun/xremap/archive/refs/tags/v${version}.tar.gz"
+checksum=323cc9a900e7bc226008179619e49ee679caa6c9ade85e69bb5e710ffb0ed7ff
+
+vlicense() {
+	vlicense LICENSE
+}
```
Comment (meator) [inline srcpkgs/xremap/template:16]: `vlicense()` isn't a function recognized by `xbps-src`. If you would try to build this package yourself, you would see:
```
/home/coder/git/void-packages/srcpkgs/xremap/template: line 14: syntax error near unexpected token `('
/home/coder/git/void-packages/srcpkgs/xremap/template: line 14: `vlicense() {'
=> ERROR: xbps-src: failed to read /home/coder/git/void-packages/srcpkgs/xremap/template!
```
```suggestion
post_install() {
	vlicense LICENSE
}
```

**Review by meator — CHANGES_REQUESTED** [review]

**Comment by tranzystorek-io** [comment]
What about the crate features (`gnome`, `sway`, `x11`)?

Also, please generate and install shell completions

**Comment by meator** [comment]
> Also, please generate and install shell completions

This won't look pretty when you take cross compiling into consideration. I know, I have dealt with these issues before. There are four ways this can be solved (sorted by their usefulness):
1. Ask upstream to distribute completion scripts in their own files in the repo instead of embedding them into the program (it seems that they're using some automatic system that generates the completions but distributing the files separately is way more practical in my opinion). You could then simply `vcompletion` these files in `post_install()`.

2. Add `xremap` to `hostmakedepends` when `$CROSS_BUILD`ing. I know, this seems strange. `xremap` depending on `xremap`? This won't create an infinite loop because this dependency is active only if `$CROSS_BUILD` is defined. This dependency means that the cross compiled version (the version that's in `$XBPS_TARGET_MACHINE` and that `xbps-src` can't directly execute) depends on the natively compiled version (the `$XBPS_MACHINE` architecture which can be executed) of `xremap`. There are two things you can do with the native version of `xremap`:
   1. Steal the completion from the native version.

      You could generate the completions normally when you aren't cross compiling and then "steal" these completions from the native package when you are cross compiling (copy them from `/usr/share/<the three shell dependent completion directories>` which was populated by the native package into `$DESTDIR`). I do not know how would the Void maintainers think of this but this seems like the simplest solution. This would normally be unacceptable, but since shell completions aren't architecture dependent, this shouldn't matter.

   2. Generate the completions with the native version.

      You could run `xremap --completions <the shells>` with the **native** version and put the completions in the appropriate locations in `$DESTDIR` (this would probably require some temporary files if you would want to use `vcompletion`). This is a bit similar to 3a9aeacde03485588f78984c4baf040ab003dd32.

   This solution is good because the completions would be automatically retrieved when an update changes them (which isn't true for the third solution).
3. Make a `srcpkgs/xremap/files` (i.e. `$FILESDIR`) directory, manually generate the files and put them there (and add them to this PR). Then `post_install()` would `vcompletion` these files.

   I don't really like this solution because every time the completions would change, the files in `$FILESDIR` would have to be updated.
4. Do not include the completions in the package. You might not call this a "solution" but it is the simplest thing to do.

If you have any questions than please comment on this.

What do you think @tranzystorek-io?

**Comment by tranzystorek-io** [comment]
I'm not sure I understand the problem. In these cases we just run `vtargetrun` with the built binary in `post_build` to generate the scripts, we do this already in many other packages.

**Comment by gbrlsnchs** [comment]
> _I didn't know that._ Well downloading 300MB of qemu just to get completions seems like a solution. My comment is pretty useless now. This is an interesting way to solve this.

Just for reference, [here's an example](https://github.com/void-linux/void-packages/blob/c8aa5e65f18faaf1f3b3fcb7dc237cd5aa8d118f/srcpkgs/github-cli/template#L30) (GitHub CLI).

**Comment by akhiljalagam** [comment]
> > _I didn't know that._ Well downloading 300MB of qemu just to get completions seems like a solution. My comment is pretty useless now. This is an interesting way to solve this.
> 
> Just for reference, [here's an example](https://github.com/void-linux/void-packages/blob/c8aa5e65f18faaf1f3b3fcb7dc237cd5aa8d118f/srcpkgs/github-cli/template#L30) (GitHub CLI).

thanks. added shell completions.

File: srcpkgs/xremap/template [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+maintainer="akhil <akhil.j@outlook.in>"
+license="MIT"
+homepage="https://github.com/k0kubun/xremap"
+changelog="https://raw.githubusercontent.com/k0kubun/xremap/master/CHANGELOG.md"
+distfiles="https://github.com/k0kubun/xremap/archive/refs/tags/v${version}.tar.gz"
+checksum=323cc9a900e7bc226008179619e49ee679caa6c9ade85e69bb5e710ffb0ed7ff
+
+post_build() {
+	xremap=$(find $GOPATH/bin -name xremap)
+	for shell in bash elvish fish powershell zsh; do
```
Comment (tranzystorek-io) [inline srcpkgs/xremap/template:16]: We can only really install bash, fish and zsh completions

Comment (akhiljalagam) [reply srcpkgs/xremap/template:16]: why?  
xremaps can generate all. 
```
        --completions <SHELL>    Generate shell completions [possible values: bash, elvish, fish,
                                 powershell, zsh]
```

Comment (akhiljalagam) [reply srcpkgs/xremap/template:16]: oops. powershell is for windows. i got it.

Comment (tranzystorek-io) [reply srcpkgs/xremap/template:16]: `vcompletion` only accepts those three, see in the manual

**Comment by meator** [comment]
You are missing `build_helper=qemu`. Please try to compile the programs yourself with `./xbps-src pkg <pkgname>` before posting them. It errors with:
```
=> xremap-0.4.4_1: running post_build ...
/void-packages/srcpkgs/xremap/template: line 17: vtargetrun: command not found
=> ERROR: xremap-0.4.4_1: post_build: 'vtargetrun $xremap --completions $shell > xremap.$shell' exited with 127
=> ERROR:   in post_build() at srcpkgs/xremap/template:17
```

**Pull request closed without merging**